code/modules/bluespace_exploration/encounter/shiploader.dm

```diff
@@ -0,0 +1,76 @@
+/datum/map_template/shuttle/ship
+	var/faction = /datum/faction/station
+
+	var/allowed_weapons = list()
+
+	var/id = null
+	var/difficulty = 0
+
+	prefix = "_maps/shuttles/exploration/"
+	suffix = null
+
+	can_be_bought = FALSE
+
+	var/amount_left = INFINITY
+
+/datum/map_template/shuttle/ship/New()
+	if(!islist(faction))
+		faction = subtypesof(faction)
+	. = ..()
+
+/datum/map_template/shuttle/ship/proc/can_place()
+	return amount_left > 0
+
+/datum/map_template/shuttle/ship/proc/try_to_place(z,allowed_areas)
+	//To give unique ids to all the spawned ships
+	var/static/shuttles_spawned = 0
+	var/sanity = PLACEMENT_TRIES
+	while(sanity > 0)
+		sanity--
+		var/width_border = TRANSITIONEDGE + SPACERUIN_MAP_EDGE_PAD + round(width / 2)
+		var/height_border = TRANSITIONEDGE + SPACERUIN_MAP_EDGE_PAD + round(height / 2)
```

Comment:
You know what to do.

Reply:
surely this would just bitshift left but ok